test/katex-spec.js

```diff
@@ -1590,31 +1590,38 @@ describe("An HTML font tree-builder", function() {
 
     it("should render \\textit{R} with the correct font", function() {
         const markup = katex.renderToString("\\textit{R}");
-        expect(markup).toContain("<span class=\"mord textit\">R</span>");
+        expect(markup).toContain("<span class=\"mord textit mathrm\">R</span>");
     });
 
     it("should render \\text{\\textit{R}} with the correct font", function() {
         const markup = katex.renderToString("\\text{\\textit{R}}");
-        expect(markup).toContain("<span class=\"mord textit\">R</span>");
+        expect(markup).toContain("<span class=\"mord textit mathrm\">R</span>");
     });
 
     it("should render \\text{R\\textit{S}T} with the correct fonts", function() {
         const markup = katex.renderToString("\\text{R\\textit{S}T}");
         expect(markup).toContain("<span class=\"mord mathrm\">R</span>");
-        expect(markup).toContain("<span class=\"mord textit\">S</span>");
+        expect(markup).toContain("<span class=\"mord textit mathrm\">S</span>");
         expect(markup).toContain("<span class=\"mord mathrm\">T</span>");
     });
 
     it("should render \\textbf{R} with the correct font", function() {
         const markup = katex.renderToString("\\textbf{R}");
-        expect(markup).toContain("<span class=\"mord mathbf\">R</span>");
+        expect(markup).toContain("<span class=\"mord textbf mathrm\">R</span>");
     });
 
     it("should render \\textsf{R} with the correct font", function() {
         const markup = katex.renderToString("\\textsf{R}");
         expect(markup).toContain("<span class=\"mord mathsf\">R</span>");
     });
 
+    it("should render \\textsf{\\textit{R}G\\textbf{B}} with the correct font", function() {
+        const markup = katex.renderToString("\\textsf{\\textit{R}G\\textbf{B}}");
+        expect(markup).toContain("<span class=\"mord textit mathsf\">R</span>");
+        expect(markup).toContain("<span class=\"mord mathsf\">G</span>");
+        expect(markup).toContain("<span class=\"mord textbf mathsf\">B</span>");
+    });
```

Comment:
You might want to try writing snapshot tests for new tests that verify things about the HTML markup we're producing.  See mathml-spec.js for examples.

Reply:
It would be good to have some tests that compare the markup produced for things that should be similar, e.g.

I would expect
```
\textsf{\textbf{mathrm{A}}}
```
to produce the same markup as
```
\mathrml{A}
```

A little bit more complicated but
```
\textsf{\textbf{\mathrm{\textsf{A}}}}
```
is equivalent to
```
\textsf{\textbf{A}}
```
I checked using quicklatex.com which I believe calls actual LaTeX on its backend.

Reply:
I thought the HTML trees rendered would be equivalent between `\textsf{\textbf{$\mathrm{A}$}}` and `\mathrm{A}`, but they're not b/c the first one has extra spans.  It's just that the visual output looks the same... unfortunately that's hard to test.  :(